comment/api/permissions.py

```diff
@@ -36,3 +36,17 @@ def has_permission(self, request, view):
 
     def has_object_permission(self, request, view, obj):
         return obj.is_flagged and (is_comment_admin(request.user) or is_comment_moderator(request.user))
+
+
+class SubscriptionEnabledPermission(permissions.BasePermission):
+    """
+    This will check if the COMMENT_ALLOW_SUBSCRIPTION is enabled
+    """
+    def has_permission(self, request, view):
+        return getattr(settings, 'COMMENT_ALLOW_SUBSCRIPTION', False)
+
+
+class GetSubscribersPermission(permissions.BasePermission):
+    def has_permission(self, request, view):
+        is_system_enabled = getattr(settings, 'COMMENT_ALLOW_SUBSCRIPTION', False)
+        return (is_comment_admin(request.user) or is_comment_moderator(request.user)) and is_system_enabled
```

Comment:
Shouldn't the first check be for `is_system_enabled`? In the present state, it will make unnecessary queries to the database when this setting in not used.